tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/IbmMq/IbmMqHelper.cs

```diff
@@ -0,0 +1,139 @@
+// <copyright file="IbmMqHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.IbmMq;
+
+internal static class IbmMqHelper
+{
+    private const string MessagingType = "ibmmq";
+
+    internal static Scope? CreateProducerScope(Tracer tracer, IMqQueue queue, IMqMessage message)
+    {
+        Scope? scope = null;
+
+        try
+        {
+            var settings = tracer.Settings;
+            if (!settings.IsIntegrationEnabled(IntegrationId.IbmMq))
+            {
+                return null;
+            }
+
+            var operationName = tracer.CurrentTraceSettings.Schema.Messaging.GetOutboundOperationName(MessagingType);
+            var serviceName = tracer.CurrentTraceSettings.Schema.Messaging.GetServiceName(MessagingType);
+
+            scope = tracer.StartActiveInternal(
+                operationName,
+                serviceName: serviceName,
+                finishOnClose: true);
+            tracer.TracerManager.Telemetry.IntegrationGeneratedSpan(IntegrationId.IbmMq);
+
+            var resourceName = $"Produce Topic {queue.Name}";
+
+            var span = scope.Span;
+            span.Type = SpanTypes.Queue;
+            span.ResourceName = resourceName;
```

Comment:
Do we not want to add the queue etc to the span Tags? 🤔 e.g. see `KafkaTags`, we add queue etc to those, I feel like we should add them here too?